common/src/main/java/org/opensearch/ml/common/connector/functions/postprocess/RemoteMlCommonsPassthroughPostProcessFunction.java

```diff
@@ -0,0 +1,117 @@
+
+import org.opensearch.ml.common.output.model.MLResultDataType;
+import org.opensearch.ml.common.output.model.ModelTensor;
+
+/**
+ * A post-processing function for calling a remote ml commons instancethat preserves the original neural sparse response structure
+ * to avoid double-wrapping when receiving responses from another ML-Commons instance.
+ */
+public class RemoteMlCommonsPassthroughPostProcessFunction extends ConnectorPostProcessFunction<Map<String, Object>> {
+    @Override
+    public void validate(Object input) {
+        if (!(input instanceof Map) && !(input instanceof List)) {
+            throw new IllegalArgumentException("Post process function input must be a Map or List");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<ModelTensor> process(Map<String, Object> input, MLResultDataType dataType) {
+        // Check if this is an ML-Commons response with inference_results
+        if (input.containsKey("inference_results") && input.get("inference_results") instanceof List) {
+            List<Map<String, Object>> inferenceResults = (List<Map<String, Object>>) input.get("inference_results");
+
+            List<ModelTensor> modelTensors = new ArrayList<>();
+            for (Map<String, Object> result : inferenceResults) {
+                // Extract the output field which contains the ModelTensor data
+                if (result.containsKey("output") && result.get("output") instanceof List) {
+                    List<Map<String, Object>> outputs = (List<Map<String, Object>>) result.get("output");
+                    for (Map<String, Object> output : outputs) {
+                        // This inner map should represent a model tensor, so we try to parse and instantiate a new one.
+                        ModelTensor modelTensor = createModelTensorFromMap(output);
+                        modelTensors.add(modelTensor);
+                    }
+                }
+            }
+
+            return modelTensors;
+        }
+
+        // Fallback for non-ML-Commons responses
+        ModelTensor tensor = ModelTensor.builder().name("response").dataAsMap(input).build();
+
+        return List.of(tensor);
+    }
+
+    /**
+     * Creates a ModelTensor from a Map<String, Object> representation based on the API format
+     * of the /_predict API
+     */
+    @SuppressWarnings("unchecked")
+    private ModelTensor createModelTensorFromMap(Map<String, Object> map) {
+        String name = (String) map.getOrDefault(ModelTensor.NAME_FIELD, OUTPUT_FIELD);
+        String result = (String) map.get(ModelTensor.RESULT_FIELD);
+        Map<String, Object> dataAsMap = (Map<String, Object>) map.get(ModelTensor.DATA_AS_MAP_FIELD);
+
+        // Handle data type
+        MLResultDataType dataType = null;
+        if (map.containsKey(ModelTensor.DATA_TYPE_FIELD)) {
+            Object dataTypeObj = map.get(ModelTensor.DATA_TYPE_FIELD);
+            if (dataTypeObj instanceof String) {
```

Comment:
nitpick: There are multiple unchecked type casting operations in this method. I think it worth adding type safety to avoid too much "unchecked" warnings.